Client/Impl/Worker/JobWorker.cs

```diff
@@ -77,10 +86,95 @@ internal void Open()
         {
             isRunning = true;
             var cancellationToken = source.Token;
-            var jobWorkerSignal = new JobWorkerSignal();
 
-            StartPollerThread(jobWorkerSignal, cancellationToken);
-            StartHandlerThreads(jobWorkerSignal, cancellationToken);
+            activateJobsCommand = jobWorkerBuilder.Command;
+            maxJobsActive = jobWorkerBuilder.Command.Request.MaxJobsToActivate;
+
+            // Create options for blocks
+            var bufferOptions = new DataflowBlockOptions
+            {
+                CancellationToken = cancellationToken,
+                EnsureOrdered = false
+            };
+
+            var executionOptions = new ExecutionDataflowBlockOptions
+            {
+                MaxDegreeOfParallelism = jobWorkerBuilder.ThreadCount,
+                CancellationToken = cancellationToken,
+                EnsureOrdered = false
+            };
+
+            // Buffer for polled jobs
+            var input = new BufferBlock<IJob>(bufferOptions);
+
+            // Transformblock to process polled jobs
+            var transformer = new TransformBlock<IJob, IJob>(async activatedJob =>
+                {
+                    try
+                    {
+                        await jobHandler(jobClient, activatedJob);
+                        await TryToAutoCompleteJob(activatedJob);
+                    }
+                    catch (Exception exception)
+                    {
+                        await FailActivatedJob(activatedJob, cancellationToken, exception);
+                    }
+                    finally
+                    {
+                        jobClient.Reset();
+                    }
+
+                    return activatedJob;
+                },
+                executionOptions);
+
+            // Action block to finalize handled tasks
+            var output = new ActionBlock<IJob>(activatedJob => { currentJobsActive--; },
+                executionOptions);
+
+            // Link blocks
+            input.LinkTo(transformer);
+            transformer.LinkTo(output);
+
+            // Start polling
+            var pollingTask = Task.Run(async () =>
```

Comment:
Would be cool if we could split the large method in smaller ones, like setup block, start polling etc.